src/lib/query-store.ts

```diff
@@ -10,6 +10,10 @@ export function pageQueryStore<T extends object>(props: T) {
    let initialLoad = true;
    page.subscribe((p) => {
       let query = queryString.parse(p.query.toString(), { parseNumbers: true });
+
+      // set the default page to 1 after navigation if the query params do not contain the page parameter and the original props contained this field
+      if ('page' in props && !query?.page) query.page = 1;
```

Comment:
Not the most elegant solution, but I can't think of a much better way to do it so happy to accept this now and maybe change it up at a later time.

Reply:
Yeah, especially since the generic store needs to know the implementation details of each type.